crates/ty_server/src/server/api/requests/completion.rs

```diff
@@ -69,3 +71,42 @@ impl BackgroundDocumentRequestHandler for CompletionRequestHandler {
 impl RetriableRequestHandler for CompletionRequestHandler {
     const RETRY_ON_CANCELLATION: bool = true;
 }
+
+fn kind(ty: Type<'_>) -> Option<CompletionItemKind> {
+    // ref https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItemKind
+    Some(match ty {
+        Type::FunctionLiteral(_)
+        | Type::DataclassDecorator(_)
+        | Type::DataclassTransformer(_)
+        | Type::Callable(_) => CompletionItemKind::FUNCTION,
+        Type::BoundMethod(_) => CompletionItemKind::METHOD,
+        Type::ModuleLiteral(_) => CompletionItemKind::MODULE,
+        Type::ClassLiteral(_) | Type::GenericAlias(_) | Type::SubclassOf(_) => {
+            CompletionItemKind::CLASS
+        }
+        // This is a little weird for "struct." I'm mostly interpreting
+        // "struct" here as a more general "object." ---AG
+        Type::NominalInstance(_)
+        | Type::ProtocolInstance(_)
+        | Type::PropertyInstance(_)
+        | Type::Tuple(_)
+        | Type::BoundSuper(_) => CompletionItemKind::STRUCT,
+        Type::IntLiteral(_) | Type::BooleanLiteral(_) | Type::TypeIs(_) => {
+            CompletionItemKind::VALUE
+        }
+        Type::StringLiteral(_) | Type::LiteralString | Type::BytesLiteral(_) => {
+            CompletionItemKind::TEXT
+        }
+        Type::TypeVar(_) => CompletionItemKind::TYPE_PARAMETER,
+        Type::Dynamic(_)
+        | Type::Never
+        | Type::MethodWrapper(_)
+        | Type::WrapperDescriptor(_)
```

Comment:
Should these be `FUNCTION` instead?

Reply:
`WrapperDescriptor` should be `FUNCTION`, `MethodWrapper` should be `METHOD`